### PR TITLE
PW-930 finalise checkout component

### PR DESCRIPTION
**Pull request by cyattilakiss**
**Description**
Style holder name input field
Shareable checkout component for oneclick
Move onValid and onError to OnChange

### CODE REVIEW

File: view/frontend/web/js/view/payment/method-renderer/adyen-cc-method.js [hunk 1]
```diff
@@ -109,8 +108,16 @@ define(
                     groupTypes: self.getAvailableCardTypeAltCodes(),
 
                     onChange: function (state) {
-                        // isValid is not present on start
-                        if (typeof state.isValid !== 'undefined' && state.isValid === false) {
+                        if (!!state.isValid) {
```
Comment (rikterbeek) [inline view/frontend/web/js/view/payment/method-renderer/adyen-cc-method.js]: why !! ?

File: view/frontend/web/js/view/payment/method-renderer/adyen-oneclick-method.js [hunk 2]
```diff
@@ -100,6 +100,20 @@ define(
             getAdyenBillingAgreements: function () {
                 var self = this;
 
+                // shareable adyen checkout component
+                var checkout = new AdyenCheckout({
```
Comment (rikterbeek) [inline view/frontend/web/js/view/payment/method-renderer/adyen-oneclick-method.js]: how is this shared ?

Comment (rikterbeek) [reply view/frontend/web/js/view/payment/method-renderer/adyen-oneclick-method.js]: never mind is clear. Agree better like this.

Comment (cyattilakiss) [reply view/frontend/web/js/view/payment/method-renderer/adyen-oneclick-method.js]: All the hpp payment methods that we render in the `_.map(window.checkoutConfig.payment.adyenOneclick.billingAgreements, function (value) {` van use this component

File: view/frontend/web/js/view/payment/method-renderer/adyen-cc-method.js (hunk 1 above)
Comment (cyattilakiss) [reply view/frontend/web/js/view/payment/method-renderer/adyen-cc-method.js]: With this check we can do the same as we did previously with the `typeof state.isValid !== 'undefined' && state.isValid === false`

Comment (cyattilakiss) [reply view/frontend/web/js/view/payment/method-renderer/adyen-cc-method.js]: TBH I learned it from the checkout guys, tested it and really does the same

**Pull request merged**